app/ts/App.tsx

```diff
@@ -321,6 +362,17 @@ export function App() {
 		universeInfo(maybeReadClient.deepValue, universe.deepValue)
 	})
 
+	useSignalEffect(() => {
+		if (reputationTokenAddress.deepValue === undefined) return
+		if (maybeReadClient.deepValue === undefined) return
+		const updateForkValues = async () => {
+			if (reputationTokenAddress.deepValue === undefined) return
+			if (maybeReadClient.deepValue === undefined) return
+			forkValues.deepValue = await getForkValues(maybeReadClient.deepValue, reputationTokenAddress.deepValue)
+		}
+		updateForkValues()
```

Comment:
This should make it so TS doesn't complain about possibly undefined values since you already check them in the outer function.
```suggestion
		getForkValues(maybeReadClient.deepValue, reputationTokenAddress.deepValue)
			.then(x => forkValues.deepValue = x)
```

Reply:
You may even be able to do `.then(forkValues.deepValue.set)`, I forget how `deepValue` is implemented and if JS lets you pass setters as functions.

Reply:
You also need a `.catch(...)` (or a `try/catch` if you stick with the full async function pattern).  See comment below for more details.